### PR TITLE
[main][feat]: claim at mini fairlaunch

### PR DESCRIPTION
**Pull request by abc-alpaca**
## Description
- add test for supporting miniFL

## ChangeLogs:
- [env] update env.example 
- [contract for test] add mock MiniFL in contract for test 
- [test] remove calculation part alpaca reward part (using mockMiniFL)

### CODE REVIEW

File: contracts/8.10/protocol/DeltaNeutralVault02.sol [hunk 1]
```diff
@@ -0,0 +1,887 @@
...
+import "./interfaces/IWNativeRelayer.sol";
+import "./interfaces/IDeltaNeutralVaultConfig.sol";
+import "./interfaces/IMiniFL.sol";
+import "./interfaces/ISwapRouter.sol";
+
+import "../utils/SafeToken.sol";
+import "../utils/FixedPointMathLib.sol";
+import "../utils/Math.sol";
+import "../utils/FullMath.sol";
+import "hardhat/console.sol";
```
Comment (jr-alpaca) [inline contracts/8.10/protocol/DeltaNeutralVault02.sol:35]: please remove import console.sol

File: contracts/8.10/protocol/interfaces/IMiniFL.sol [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
```
Comment (jr-alpaca) [inline contracts/8.10/protocol/interfaces/IMiniFL.sol:1]:
```suggestion
// SPDX-License-Identifier: BUSL
```

File: contracts/8.10/token/interfaces/IRewarder.sol [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
```
Comment (jr-alpaca) [inline contracts/8.10/token/interfaces/IRewarder.sol:1]:
```suggestion
// SPDX-License-Identifier: BUSL
```

File: contracts/8.10/token/interfaces/IMiniFL.sol [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
```
Comment (jr-alpaca) [inline contracts/8.10/token/interfaces/IMiniFL.sol:1]:
```suggestion
// SPDX-License-Identifier: BUSL
```

File: .env.example [hunk 5]
```diff
@@ -1,16 +1,18 @@
 # Testnet Env
 BSC_TESTNET_PRIVATE_KEY="xxx"
 ETH_TESTNET_PRIVATE_KEY="xxx"
+FANTOM_TESTNET_PRIVATE_KEY="xxx"
```
Comment (sora-alpaca) [inline .env.example:4]: should it be FANTOM or FTM

File: contracts/8.10/protocol/DeltaNeutralVault02.sol [hunk 6]
```diff
@@ -0,0 +1,886 @@
...
+
+    if (reinvestPath.length == 0) {
+      revert DeltaNeutralVault_BadReinvestPath();
+    }
+
+    // 1.  claim reward from fairlaunch
+    uint256 _equityBefore = totalEquityValue();
+    uint256 _alpacaBefore = IERC20Upgradeable(alpacaToken).balanceOf(address(this));
+
+    address _fairLaunchAddress = config.fairLaunchAddr();
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVault02.sol:469]:
```suggestion
    address _miniFLAddress= config.fairLaunchAddr();
```

File: contracts/8.10/protocol/DeltaNeutralVault02.sol [hunk 7]
```diff
@@ -0,0 +1,886 @@
...
+    IMiniFL(_fairLaunchAddress).harvest(IVault(stableVault).fairLaunchPoolId());
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVault02.sol:470]:
```suggestion
    IMiniFL(_miniFLAddress).harvest(IVault(stableVault).fairLaunchPoolId());
```

File: contracts/8.10/protocol/DeltaNeutralVault02.sol [hunk 8]
```diff
@@ -0,0 +1,886 @@
...
+    IMiniFL(_fairLaunchAddress).harvest(IVault(assetVault).fairLaunchPoolId());
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVault02.sol:471]:
```suggestion
    IMiniFL(_miniFLAddress).harvest(IVault(assetVault).fairLaunchPoolId());
```

**Review by alpanaca — CHANGES_REQUESTED** [review]

File: .env.example (hunk 5 above)
Comment (alpanaca) [reply .env.example:4]: Should be any but keep it consistent.

Comment (abc-alpaca) [reply .env.example:4]: FANTOM is correct

**Review by alpanaca — DISMISSED** [review]

File: contracts/8.10/protocol/DeltaNeutralVault02.sol [hunk 9]
```diff
@@ -0,0 +1,886 @@
...
+import "../utils/SafeToken.sol";
+import "../utils/FixedPointMathLib.sol";
+import "../utils/Math.sol";
+import "../utils/FullMath.sol";
+
+/// @title Delta Neutral Vault is designed to take a long and short position in an asset at the same time
+/// to cancel out the effect on the out-standing portfolio when the asset’s price moves.
+/// @notice contract for ftm environment
+// solhint-disable max-states-count
+contract DeltaNeutralVault02 is ERC20Upgradeable, ReentrancyGuardUpgradeable, OwnableUpgradeable {
```
Comment (spicysquid168) [inline contracts/8.10/protocol/DeltaNeutralVault02.sol:40]: I don't think we need to add `DeltaNeutralVault02` at all. The only diff is just importing the interface of liquidity mining contract which in fact no need to be changed as the function that need to make a call to liquidity mining is fully compatible in `DeltaNeutralVault` as well

File: contracts/8.10/protocol/interfaces/IMiniFL.sol [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  ξ　~　ξ 
+  ξ　　 ξ 
+  ξ　　 “~～~～〇 
+  ξ　　　　　　 ξ 
+  ξ ξ ξ~～~ξ ξ ξ 
+　 ξ_ξξ_ξ　ξ_ξξ_ξ
+Alpaca Fin Corporation
+*/
+
+pragma solidity 0.8.10;
```
Comment (spicysquid168) [inline contracts/8.10/protocol/interfaces/IMiniFL.sol:14]: This also no need

**Comment by spicysquid168** [comment]
I think the only that need to be done in this PR is to write a test on `DeltaNeutralVault` + `MiniFL`. Other parts can be left out.

**Review by spicysquid168 — CHANGES_REQUESTED** [review]

**Review by spicysquid168 — APPROVED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**